src/index.js

```diff
@@ -124,16 +124,16 @@ export function CartProvider({
   const addItem = (item, quantity = 1) => {
     if (quantity <= 0) return;
     if (!item.id) throw new Error("You must provide an `id` for items");
-
+    if (!item.price) throw new Error("You must pass a `price` for new items");
+    
     const currentItem = state.items.find((i) => i.id === item.id);
 
-    if (!currentItem & !item.price)
-      throw new Error("You must pass a `price` for new items");
-
     if (!currentItem) {
       const payload = { ...item, quantity };
+      dispatch({ type: ADD_ITEM, payload });
+      
       onItemAdd && onItemAdd(payload);
-      return dispatch({ type: ADD_ITEM, payload });
+      return;
```

Comment:
I'm curious on the behaviour change here. Can you explain a little more about what's changed and why @halafy-ds ?

Reply:
This change is just for calling dispatch before onItemAdd, firstly to keep consistency across all cases (Add, Update and Remove) and secondly because im triying to make localStorage optional and i'm experiencing some wird behaviour when tring to cal onItemAdd, onItemUpdate.. wuth a custom fackestorage object. It's more a guess tha other thing and as it doesn't hurt i'm suggesting it.